cmd/artifact_upload.go

```diff
@@ -86,15 +94,29 @@ func NewArtifactUploadCmd(cmd *cobra.Command, args []string) (*ArtifactUploadCmd
 		artifactPath:    args[0],
 		skipVerify:      skipVerify,
 		withoutProgress: withoutProgress,
+		direct:          direct,
 	}, nil
 }
 
 func (c *ArtifactUploadCmd) Run() error {
-
 	client := deployments.NewClient(c.server, c.skipVerify)
-	err := client.UploadArtifact(c.description, c.artifactPath, c.token, c.withoutProgress)
-	if err != nil {
-		return err
+	if c.direct {
+		log.Infof("getting direct link.\n")
+		link, err := client.DirectDownloadLink(c.token)
+		if err != nil {
+			return errors.Wrap(err, "failed to get the direct pre-signed URL")
+		}
+
+		log.Infof("uploading the artifact.\n")
+		err = client.DirectUpload(c.artifactPath, c.token, link.Uri, c.withoutProgress)
+		if err != nil {
+			return errors.Wrap(err, "failed to upload the artifact")
+		}
```

Comment:
TODO: Need to call the `/complete` endpoint.

Reply:
yes, I commented on that one.

Reply:
added.

Reply:
and btw @alfrunes token returned as parameter :) I knew I wanted it for something.